tests/api/api/test_artifacts.py

```diff
@@ -82,3 +87,36 @@ def test_delete_artifacts_after_storing_empty_dict(db: Session, client: TestClie
 
     resp = client.get(project_artifacts_path)
     assert len(resp.json()["artifacts"]) == 0
+
+
+def test_list_artifacts(db: Session, client: TestClient) -> None:
+    _create_project(client)
+
+    for artifact_path in [
+        f"{LEGACY_API_ARTIFACTS_PATH}?project={PROJECT}",
+        API_ARTIFACTS_PATH.format(project=PROJECT),
+    ]:
+        resp = client.get(artifact_path)
+        assert resp.status_code == HTTPStatus.OK.value
+        assert len(resp.json()["artifacts"]) == 0
+
+    resp = client.post(
+        f"{LEGACY_API_ARTIFACT_PATH}/{PROJECT}/{UID}/{KEY}?tag={TAG}", data="{}"
```

Comment:
That's why your new URLs worked without the `:path` selector - the `KEY` tested here is without slashes in it (it is just `some-key`). This works, but it doesn't expose the problem.